core/metacling/src/TClingTypedefInfo.cxx

```diff
@@ -296,18 +281,18 @@ const char *TClingTypedefInfo::TrueName(const ROOT::TMetaUtils::TNormalizedCtxt
 ////////////////////////////////////////////////////////////////////////////////
 /// Get the name of the current typedef.
 
-const char *TClingTypedefInfo::Name() const
+const char *TClingTypedefInfo::Name()
 {
    if (!IsValid()) {
       return "(unknown)";
    }
-   // Note: This must be static because we return a pointer to the internals.
-   TTHREAD_TLS_DECL( std::string, fullname);
-   fullname.clear();
-   const clang::TypedefNameDecl *td = llvm::dyn_cast<clang::TypedefNameDecl>(fDecl);
-   const clang::ASTContext &ctxt = fDecl->getASTContext();
-   ROOT::TMetaUtils::GetFullyQualifiedTypeName(fullname,ctxt.getTypedefType(td),*fInterp);
-   return fullname.c_str();
+   if (!fNameCache.empty())
+     return fNameCache.c_str();
+
+   const clang::TypedefNameDecl *td = llvm::cast<clang::TypedefNameDecl>(fDecl);
+   const clang::ASTContext &ctxt = GetDecl()->getASTContext();
```

Comment:
Is there a different between the value of fDecl and GetDecl() here?

Reply:
No, this class does not override `GetDecl` and the base class implementation returns `fDecl`.

Reply:
If they are the same, can we use one of the two in both line (otherwise the future reader will be puzzled :) )

Reply:
Fixed.